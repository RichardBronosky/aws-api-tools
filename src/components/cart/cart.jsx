import React from 'react'
import classNames from 'classnames'
import CondensedContact from '../shared/condensed-contact-links'
import CartProduct from './cart-parts/cart-product'
import CreditCardBanner from './cart-parts/credit-card-banner'
import CartEmpty from './cart-parts/cart-empty'
import CheckoutSticky from '../checkout/checkout-sticky/checkout-sticky'
import PaymentMethods from './cart-parts/payment-methods'
import PromotionContentGroup from '../shared/promotion-content-group'
import { removeUnavailableItems } from '../../lib/helpers/cart'
import '../../assets/css/components/cart/cart.sass'

export default ({
  cart,
  isMobile,
  onAddActiveAddon,
  onRemoveActiveAddon,
  skusNotAvailable,
  discount,
  showPayPal,
  promoTargetSkus,
  setCartState,
}) => (
  <div className="cart-page">
    <h1 id="main-cart-title" tabIndex="-1">
      CART
    </h1>
    <div
      className={ classNames('grid-x', {
        'grid-margin-x': isMobile,
      }) }
    >
      <div
        className={ classNames('cart-product-list-container small-12 medium-12 large-9 grid-x grid-padding-y', {
          cell: isMobile,
        }) }
      >
        { cart &&
          cart.cartItems &&
          cart.cartItems.map((item, index) => (
            <CartProduct
              key={ index }
              product={ item.product }
              index={ index }
              productCount={ cart.cartItems.length }
              quantity={ item.quantity }
              price={ item.price }
              unavailableItem={ skusNotAvailable.filter(prod => prod.sku === item.sku)[0] }
              activeAddons={ item.activeAddons }
              onAddActiveAddon={ onAddActiveAddon }
              onRemoveActiveAddon={ onRemoveActiveAddon }
            />
          )) }
        { (!cart || (cart && cart.cartItems && cart.cartItems.length < 1) || (cart && !cart.cartItems)) && <CartEmpty /> }
        { cart && cart.cartItems && cart.cartItems.length > 0 && (
          <>
            <PromotionContentGroup targetSkus={ promoTargetSkus } isCart />
            <div className="cart-bottom-checkout cell small-12 grid-x">
              <div className="cell large-8 payment-methods">
                <PaymentMethods />
              </div>
              <CheckoutSticky
                cart={ cart }
                productsAvailable={ skusNotAvailable.length < 1 }
                removeUnavailableItems={ () => removeUnavailableItems(setCartState, skusNotAvailable) }
                onlyButtons
                discount={ discount }
                showPayPal={ showPayPal }
              />
            </div>
          </>
        ) }
        { cart && cart.cartItems && cart.cartItems.length > 0 && (
          <>
            <p className="financing-disclaimer cell small-12">*when paying with a rooms to go credit card</p>
            <CreditCardBanner />
          </>
        ) }
        <CondensedContact />
      </div>
      <div className="checkout-container cell small-12 large-2 grid-margin-y">
        { cart && cart.cartItems && cart.cartItems.length > 0 && (
          <CheckoutSticky
            cart={ cart }
            isMobile={ isMobile }
            productsAvailable={ skusNotAvailable.length < 1 }
            removeUnavailableItems={ () => removeUnavailableItems(setCartState, skusNotAvailable) }
            rightSticky
            discount={ discount }
            showPayPal={ showPayPal }
          />
        ) }
      </div>
    </div>
  </div>
)
