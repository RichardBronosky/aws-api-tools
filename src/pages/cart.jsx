import React from 'react'
import Helmet from 'react-helmet'
import Layout from '../components/layout'
import Cart from '../components/cart'

export default ({ props }) => (
  <Layout { ...props } cartPage>
    <Helmet title={ 'Shopping Cart - Rooms To Go' } meta={ [{ name: 'robots', content: 'noindex, nofollow' }] } />
    <Cart />
  </Layout>
)
