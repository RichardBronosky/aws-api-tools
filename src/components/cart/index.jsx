import React from 'react'
import { connect } from 'react-redux'
import { getLastParamOfPath } from '../../lib/helpers/string-helper'
import { verifyAndUpdateCart, cartUpdate, getPromoTargetSkus, loadCartFromOrder } from '../../lib/helpers/cart'
import { setupAnalytics } from '../../lib/helpers/google-tag-manager'
import Cart from './cart'

let apiCalled = false

class CartWrapper extends React.Component {
  state = {
    skusNotAvailable: [],
    discount: null,
    showPayPal: false,
  }

  setCartState = state => this.setState({ ...this.state, ...state })

  setApiCalled = called => {
    apiCalled = called
  }

  componentDidMount() {
    const { cart } = this.props
    const { discount, skusNotAvailable } = this.state
    setupAnalytics({ pageData: { type: 'cart', title: 'Cart', path: '/cart' } })
    const lastParam = getLastParamOfPath()
    const pattern = new RegExp('[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}')
    if (lastParam && pattern.test(lastParam)) {
      loadCartFromOrder(lastParam)
    }
    if (JSON.stringify(cart) !== JSON.stringify({ cartItems: [] })) {
      verifyAndUpdateCart(cart)
      cartUpdate(cart, skusNotAvailable, apiCalled, discount, this.setCartState, this.setApiCalled, false)
    }
  }

  componentDidUpdate(prevProps) {
    const { cart } = this.props
    const { discount, skusNotAvailable } = this.state
    if (prevProps.cart !== cart) {
      verifyAndUpdateCart(cart)
      cartUpdate(cart, skusNotAvailable, apiCalled, discount, this.setCartState, this.setApiCalled, discount !== null)
    }
  }

  render() {
    const promoTargetSkus = getPromoTargetSkus()
    return <Cart { ...this.props } { ...this.state } promoTargetSkus={ promoTargetSkus } setCartState={ this.setCartState } />
  }
}

const mapStateToProps = state => {
  return { ...state.global, ...state.cart }
}

const mapDispatchToProps = dispatch => {
  return {
    onAddActiveAddon: (productSku, addon, cartIndex) => dispatch(addActiveAddon(productSku, addon, cartIndex)),
    onRemoveActiveAddon: (productSku, addonSku, cartIndex) =>
      dispatch(removeActiveAddon(productSku, addonSku, cartIndex)),
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CartWrapper)
