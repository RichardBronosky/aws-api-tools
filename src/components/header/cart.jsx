import React from 'react'
import { connect } from 'react-redux'
import { getCartTotal } from '../../lib/helpers/cart'
import { currencyFormatUS } from '../../lib/helpers/string-helper'
import { isProductStrikeThrough } from '../../lib/helpers/product'
import RTGLink from '../shared/link'
import RemoveFromCart from '../cart/cart-parts/remove-from-cart'
import { productUrl } from '../../lib/helpers/route'

class CartPopout extends React.Component {
  render() {
    const cart = this.props.cart
    return (
      <>
        <div className="section">
          <div id="cartSubtotal" className="subtotal">
            Subtotal: { currencyFormatUS(getCartTotal(cart)) }
          </div>
        </div>
        { (!cart || (cart && typeof cart.cartItems === 'undefined')) && (
          <>
            <div className="section">
              <div className="message">Your shopping cart is empty</div>
            </div>
            { /* <div className="section">
              <div className="notice">
                <p>
                  If you saved items on a previous visit,{ ' ' }
                  <a href="/">Sign In</a> to access your saved cart.
                  </p>
              </div>
            </div> */ }
          </>
        ) }
        <div className="section sticky">
          <RTGLink
            data={ {
              slug: '/checkout',
              title: 'Checkout',
              category: 'header',
              action: 'cart interaction',
              label: 'checkout',
            } }
          >
            <span className="blue-action-btn checkout">Checkout</span>
          </RTGLink>
          <RTGLink
            data={ {
              slug: '/cart',
              title: 'View Full Cart',
              category: 'header',
              action: 'cart interaction',
              label: 'view full cart',
            } }
            className="bold-arrow-link"
          >
            View Full Cart >
          </RTGLink>
        </div>
        { cart && cart.cartItems && cart.cartItems.length > 0 && (
          <>
            { cart.cartItems &&
              cart.cartItems.map((item, index) => {
                const product = item.product
                const strikethrough = isProductStrikeThrough(product)
                return (
                  <div key={ index } className="line-item grid-x">
                    <div className="product cell small-10">
                      <RTGLink
                        data={ {
                          url: productUrl(product.title, product.sku),
                        } }
                        aria-hidden="true"
                        tabIndex="-1"
                        disabled={ product.category === 'gift-card' }
                      >
                        <div className="prod-img">
                          <img src={ product.primary_image } alt="" role="presentation" />
                        </div>
                      </RTGLink>
                      <div className="prod-details">
                        <RTGLink
                          data={ {
                            url: productUrl(product.title, product.sku),
                          } }
                          aria-describedby={ `mcq_${ index }mcp_${ index }` }
                          disabled={ product.category === 'gift-card' }
                        >
                          <div className="title">{ product.title }</div>
                        </RTGLink>
                        <div id={ 'mcq_' + index } className="qty">
                          Qty: { item.quantity }
                        </div>
                        <div id={ 'mcp_' + index } className="price">
                          { strikethrough && (
                            <p className="strikethrough">{ currencyFormatUS(item.product.list_price) }</p>
                          ) }
                          <p className={ strikethrough ? 'strikethrough-sale' : '' }>{ currencyFormatUS(item.price) }</p>
                        </div>
                      </div>
                    </div>

                    <div className="cell small-2">
                      <RemoveFromCart headerCart product={ product } index={ index } />
                    </div>
                  </div>
                )
              }) }
          </>
        ) }
      </>
    )
  }
}

const mapStateToProps = state => {
  return { ...state.cart }
}

export default connect(
  mapStateToProps,
  null
)(CartPopout)
