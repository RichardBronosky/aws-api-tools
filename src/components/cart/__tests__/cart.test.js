import React from 'react'
import renderer from 'react-test-renderer'
import Cart from '../cart'
import * as cartHelper from '../../../lib/helpers/cart'
import { mockCart } from '../../../lib/mocks/cartDataMocks'

jest.mock('../cart-parts/cart-product', () => 'CartProduct')
jest.mock('../cart-parts/cart-empty', () => 'CartEmpty')
jest.mock('../cart-parts/payment-methods', () => 'PaymentMethods')
jest.mock('../cart-parts/credit-card-banner', () => 'CreditCardBanner')
jest.mock('../../checkout/checkout-sticky/checkout-sticky', () => 'CheckoutSticky')
jest.mock('../../shared/condensed-contact-links', () => 'CondensedContact')
jest.mock('../../shared/promotion-content-group', () => 'PromotionContentGroup')

describe('Cart', () => {
  let testProps = {
    cart: mockCart,
    discount: 0,
    skusNotAvailable: [],
    isMobile: false,
    onAddActiveAddon: jest.fn(),
    onRemoveActiveAddon: jest.fn(),
    showPayPal: false,
    promoTargetSkus: [],
    setCartState: jest.fn(),
  }

  it('renders correctly when all skus available', () => {
    const tree = renderer.create(<Cart { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when skus not available', () => {
    const tree = renderer.create(<Cart { ...{ ...testProps, skusNotAvailable: ['23122360'] } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls removeUnavailableItems in right sticky when remove is clicked', () => {
    cartHelper.removeUnavailableItems = jest.fn()
    const tree = renderer.create(<Cart { ...{ ...testProps, skusNotAvailable: ['23122360'] } } />)
    tree.root.findByProps({ rightSticky: true }).props.removeUnavailableItems()
    expect(cartHelper.removeUnavailableItems).toHaveBeenCalled()
  })

  it('calls removeUnavailableItems in right sticky when remove is clicked and is mobile', () => {
    cartHelper.removeUnavailableItems = jest.fn()
    const tree = renderer.create(<Cart { ...{ ...testProps, skusNotAvailable: ['23122360'], isMobile: true } } />)
    tree.root.findByProps({ onlyButtons: true }).props.removeUnavailableItems()
    expect(cartHelper.removeUnavailableItems).toHaveBeenCalled()
  })

  it('renders correctly when cart is empty', () => {
    const tree = renderer.create(<Cart { ...{ ...testProps, cart: null } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
