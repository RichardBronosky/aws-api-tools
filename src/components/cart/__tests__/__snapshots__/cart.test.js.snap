// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders correctly when all skus available 1`] = `
<div
  className="cart-page"
>
  <h1
    id="main-cart-title"
    tabIndex="-1"
  >
    CART
  </h1>
  <div
    className="grid-x"
  >
    <div
      className="cart-product-list-container small-12 medium-12 large-9 grid-x grid-padding-y"
    >
      <CartProduct
        activeAddons={null}
        index={0}
        onAddActiveAddon={[MockFunction]}
        onRemoveActiveAddon={[MockFunction]}
        price={279.99}
        product={
          Object {
            "category": "tables",
            "delivery_type": "O",
            "free_shipping": true,
            "list_price": 349.99,
            "on_sale": true,
            "price": 279.99,
            "primary_image": "http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090",
            "shipping_cost_code": Object {
              "FL": "VD71",
              "SE": "VD71",
              "TX": "VD71",
            },
            "sku": "23122360",
            "strikethrough": Object {
              "FL": true,
              "OM": true,
              "SE": true,
              "TX": true,
            },
            "title": "Adaire Brown Side Table",
          }
        }
        productCount={2}
        quantity={1}
      />
      <CartProduct
        activeAddons={null}
        index={1}
        onAddActiveAddon={[MockFunction]}
        onRemoveActiveAddon={[MockFunction]}
        price={1199.99}
        product={
          Object {
            "category": "livingroom",
            "delivery_type": "D",
            "free_shipping": false,
            "list_price": 1299.99,
            "on_sale": true,
            "package_skus": Object {
              "FL": Array [
                Object {
                  "price": "248.65",
                  "quantity": "1",
                  "sku": "12316762",
                },
                Object {
                  "price": "185.62",
                  "quantity": "4",
                  "sku": "12116768",
                },
                Object {
                  "price": "223.71",
                  "quantity": "1",
                  "sku": "12216760",
                },
                Object {
                  "price": "73.16",
                  "quantity": "1",
                  "sku": "12516766",
                },
              ],
              "SE": Array [
                Object {
                  "price": "208.43",
                  "quantity": 1,
                  "sku": "12216760",
                },
                Object {
                  "price": "172.94",
                  "quantity": 4,
                  "sku": "12116768",
                },
                Object {
                  "price": "231.63",
                  "quantity": 1,
                  "sku": "12316762",
                },
                Object {
                  "price": "68.17",
                  "quantity": 1,
                  "sku": "12516766",
                },
              ],
              "TX": Array [
                Object {
                  "price": "231.63",
                  "quantity": "1",
                  "sku": "12316762",
                },
                Object {
                  "price": "208.43",
                  "quantity": "1",
                  "sku": "12216760",
                },
                Object {
                  "price": "68.17",
                  "quantity": "1",
                  "sku": "12516766",
                },
                Object {
                  "price": "172.94",
                  "quantity": "4",
                  "sku": "12116768",
                },
              ],
            },
            "price": 1199.99,
            "primary_image": "http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc",
            "shipping_cost_code": Object {
              "FL": "ROOM",
              "SE": "ROOM",
              "TX": "ROOM",
            },
            "sku": "1141677P",
            "title": "Laney Park Dark Gray 7 Pc Sectional",
          }
        }
        productCount={2}
        quantity={3}
      />
      <PromotionContentGroup
        isCart={true}
        targetSkus={Array []}
      />
      <div
        className="cart-bottom-checkout cell small-12 grid-x"
      >
        <div
          className="cell large-8 payment-methods"
        >
          <PaymentMethods />
        </div>
        <CheckoutSticky
          cart={
            Object {
              "cartItems": Array [
                Object {
                  "activeAddons": null,
                  "price": 279.99,
                  "product": Object {
                    "category": "tables",
                    "delivery_type": "O",
                    "free_shipping": true,
                    "list_price": 349.99,
                    "on_sale": true,
                    "price": 279.99,
                    "primary_image": "http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090",
                    "shipping_cost_code": Object {
                      "FL": "VD71",
                      "SE": "VD71",
                      "TX": "VD71",
                    },
                    "sku": "23122360",
                    "strikethrough": Object {
                      "FL": true,
                      "OM": true,
                      "SE": true,
                      "TX": true,
                    },
                    "title": "Adaire Brown Side Table",
                  },
                  "quantity": 1,
                  "sku": "23122360",
                },
                Object {
                  "activeAddons": null,
                  "price": 1199.99,
                  "product": Object {
                    "category": "livingroom",
                    "delivery_type": "D",
                    "free_shipping": false,
                    "list_price": 1299.99,
                    "on_sale": true,
                    "package_skus": Object {
                      "FL": Array [
                        Object {
                          "price": "248.65",
                          "quantity": "1",
                          "sku": "12316762",
                        },
                        Object {
                          "price": "185.62",
                          "quantity": "4",
                          "sku": "12116768",
                        },
                        Object {
                          "price": "223.71",
                          "quantity": "1",
                          "sku": "12216760",
                        },
                        Object {
                          "price": "73.16",
                          "quantity": "1",
                          "sku": "12516766",
                        },
                      ],
                      "SE": Array [
                        Object {
                          "price": "208.43",
                          "quantity": 1,
                          "sku": "12216760",
                        },
                        Object {
                          "price": "172.94",
                          "quantity": 4,
                          "sku": "12116768",
                        },
                        Object {
                          "price": "231.63",
                          "quantity": 1,
                          "sku": "12316762",
                        },
                        Object {
                          "price": "68.17",
                          "quantity": 1,
                          "sku": "12516766",
                        },
                      ],
                      "TX": Array [
                        Object {
                          "price": "231.63",
                          "quantity": "1",
                          "sku": "12316762",
                        },
                        Object {
                          "price": "208.43",
                          "quantity": "1",
                          "sku": "12216760",
                        },
                        Object {
                          "price": "68.17",
                          "quantity": "1",
                          "sku": "12516766",
                        },
                        Object {
                          "price": "172.94",
                          "quantity": "4",
                          "sku": "12116768",
                        },
                      ],
                    },
                    "price": 1199.99,
                    "primary_image": "http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc",
                    "shipping_cost_code": Object {
                      "FL": "ROOM",
                      "SE": "ROOM",
                      "TX": "ROOM",
                    },
                    "sku": "1141677P",
                    "title": "Laney Park Dark Gray 7 Pc Sectional",
                  },
                  "quantity": 3,
                  "sku": "1141677P",
                },
              ],
            }
          }
          discount={0}
          onlyButtons={true}
          productsAvailable={true}
          removeUnavailableItems={[Function]}
          showPayPal={false}
        />
      </div>
      <p
        className="financing-disclaimer cell small-12"
      >
        *when paying with a rooms to go credit card
      </p>
      <CreditCardBanner />
      <CondensedContact />
    </div>
    <div
      className="checkout-container cell small-12 large-2 grid-margin-y"
    >
      <CheckoutSticky
        cart={
          Object {
            "cartItems": Array [
              Object {
                "activeAddons": null,
                "price": 279.99,
                "product": Object {
                  "category": "tables",
                  "delivery_type": "O",
                  "free_shipping": true,
                  "list_price": 349.99,
                  "on_sale": true,
                  "price": 279.99,
                  "primary_image": "http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090",
                  "shipping_cost_code": Object {
                    "FL": "VD71",
                    "SE": "VD71",
                    "TX": "VD71",
                  },
                  "sku": "23122360",
                  "strikethrough": Object {
                    "FL": true,
                    "OM": true,
                    "SE": true,
                    "TX": true,
                  },
                  "title": "Adaire Brown Side Table",
                },
                "quantity": 1,
                "sku": "23122360",
              },
              Object {
                "activeAddons": null,
                "price": 1199.99,
                "product": Object {
                  "category": "livingroom",
                  "delivery_type": "D",
                  "free_shipping": false,
                  "list_price": 1299.99,
                  "on_sale": true,
                  "package_skus": Object {
                    "FL": Array [
                      Object {
                        "price": "248.65",
                        "quantity": "1",
                        "sku": "12316762",
                      },
                      Object {
                        "price": "185.62",
                        "quantity": "4",
                        "sku": "12116768",
                      },
                      Object {
                        "price": "223.71",
                        "quantity": "1",
                        "sku": "12216760",
                      },
                      Object {
                        "price": "73.16",
                        "quantity": "1",
                        "sku": "12516766",
                      },
                    ],
                    "SE": Array [
                      Object {
                        "price": "208.43",
                        "quantity": 1,
                        "sku": "12216760",
                      },
                      Object {
                        "price": "172.94",
                        "quantity": 4,
                        "sku": "12116768",
                      },
                      Object {
                        "price": "231.63",
                        "quantity": 1,
                        "sku": "12316762",
                      },
                      Object {
                        "price": "68.17",
                        "quantity": 1,
                        "sku": "12516766",
                      },
                    ],
                    "TX": Array [
                      Object {
                        "price": "231.63",
                        "quantity": "1",
                        "sku": "12316762",
                      },
                      Object {
                        "price": "208.43",
                        "quantity": "1",
                        "sku": "12216760",
                      },
                      Object {
                        "price": "68.17",
                        "quantity": "1",
                        "sku": "12516766",
                      },
                      Object {
                        "price": "172.94",
                        "quantity": "4",
                        "sku": "12116768",
                      },
                    ],
                  },
                  "price": 1199.99,
                  "primary_image": "http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc",
                  "shipping_cost_code": Object {
                    "FL": "ROOM",
                    "SE": "ROOM",
                    "TX": "ROOM",
                  },
                  "sku": "1141677P",
                  "title": "Laney Park Dark Gray 7 Pc Sectional",
                },
                "quantity": 3,
                "sku": "1141677P",
              },
            ],
          }
        }
        discount={0}
        isMobile={false}
        productsAvailable={true}
        removeUnavailableItems={[Function]}
        rightSticky={true}
        showPayPal={false}
      />
    </div>
  </div>
</div>
`;

exports[`Cart renders correctly when cart is empty 1`] = `
<div
  className="cart-page"
>
  <h1
    id="main-cart-title"
    tabIndex="-1"
  >
    CART
  </h1>
  <div
    className="grid-x"
  >
    <div
      className="cart-product-list-container small-12 medium-12 large-9 grid-x grid-padding-y"
    >
      <CartEmpty />
      <CondensedContact />
    </div>
    <div
      className="checkout-container cell small-12 large-2 grid-margin-y"
    />
  </div>
</div>
`;

exports[`Cart renders correctly when skus not available 1`] = `
<div
  className="cart-page"
>
  <h1
    id="main-cart-title"
    tabIndex="-1"
  >
    CART
  </h1>
  <div
    className="grid-x"
  >
    <div
      className="cart-product-list-container small-12 medium-12 large-9 grid-x grid-padding-y"
    >
      <CartProduct
        activeAddons={null}
        index={0}
        onAddActiveAddon={[MockFunction]}
        onRemoveActiveAddon={[MockFunction]}
        price={279.99}
        product={
          Object {
            "category": "tables",
            "delivery_type": "O",
            "free_shipping": true,
            "list_price": 349.99,
            "on_sale": true,
            "price": 279.99,
            "primary_image": "http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090",
            "shipping_cost_code": Object {
              "FL": "VD71",
              "SE": "VD71",
              "TX": "VD71",
            },
            "sku": "23122360",
            "strikethrough": Object {
              "FL": true,
              "OM": true,
              "SE": true,
              "TX": true,
            },
            "title": "Adaire Brown Side Table",
          }
        }
        productCount={2}
        quantity={1}
      />
      <CartProduct
        activeAddons={null}
        index={1}
        onAddActiveAddon={[MockFunction]}
        onRemoveActiveAddon={[MockFunction]}
        price={1199.99}
        product={
          Object {
            "category": "livingroom",
            "delivery_type": "D",
            "free_shipping": false,
            "list_price": 1299.99,
            "on_sale": true,
            "package_skus": Object {
              "FL": Array [
                Object {
                  "price": "248.65",
                  "quantity": "1",
                  "sku": "12316762",
                },
                Object {
                  "price": "185.62",
                  "quantity": "4",
                  "sku": "12116768",
                },
                Object {
                  "price": "223.71",
                  "quantity": "1",
                  "sku": "12216760",
                },
                Object {
                  "price": "73.16",
                  "quantity": "1",
                  "sku": "12516766",
                },
              ],
              "SE": Array [
                Object {
                  "price": "208.43",
                  "quantity": 1,
                  "sku": "12216760",
                },
                Object {
                  "price": "172.94",
                  "quantity": 4,
                  "sku": "12116768",
                },
                Object {
                  "price": "231.63",
                  "quantity": 1,
                  "sku": "12316762",
                },
                Object {
                  "price": "68.17",
                  "quantity": 1,
                  "sku": "12516766",
                },
              ],
              "TX": Array [
                Object {
                  "price": "231.63",
                  "quantity": "1",
                  "sku": "12316762",
                },
                Object {
                  "price": "208.43",
                  "quantity": "1",
                  "sku": "12216760",
                },
                Object {
                  "price": "68.17",
                  "quantity": "1",
                  "sku": "12516766",
                },
                Object {
                  "price": "172.94",
                  "quantity": "4",
                  "sku": "12116768",
                },
              ],
            },
            "price": 1199.99,
            "primary_image": "http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc",
            "shipping_cost_code": Object {
              "FL": "ROOM",
              "SE": "ROOM",
              "TX": "ROOM",
            },
            "sku": "1141677P",
            "title": "Laney Park Dark Gray 7 Pc Sectional",
          }
        }
        productCount={2}
        quantity={3}
      />
      <PromotionContentGroup
        isCart={true}
        targetSkus={Array []}
      />
      <div
        className="cart-bottom-checkout cell small-12 grid-x"
      >
        <div
          className="cell large-8 payment-methods"
        >
          <PaymentMethods />
        </div>
        <CheckoutSticky
          cart={
            Object {
              "cartItems": Array [
                Object {
                  "activeAddons": null,
                  "price": 279.99,
                  "product": Object {
                    "category": "tables",
                    "delivery_type": "O",
                    "free_shipping": true,
                    "list_price": 349.99,
                    "on_sale": true,
                    "price": 279.99,
                    "primary_image": "http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090",
                    "shipping_cost_code": Object {
                      "FL": "VD71",
                      "SE": "VD71",
                      "TX": "VD71",
                    },
                    "sku": "23122360",
                    "strikethrough": Object {
                      "FL": true,
                      "OM": true,
                      "SE": true,
                      "TX": true,
                    },
                    "title": "Adaire Brown Side Table",
                  },
                  "quantity": 1,
                  "sku": "23122360",
                },
                Object {
                  "activeAddons": null,
                  "price": 1199.99,
                  "product": Object {
                    "category": "livingroom",
                    "delivery_type": "D",
                    "free_shipping": false,
                    "list_price": 1299.99,
                    "on_sale": true,
                    "package_skus": Object {
                      "FL": Array [
                        Object {
                          "price": "248.65",
                          "quantity": "1",
                          "sku": "12316762",
                        },
                        Object {
                          "price": "185.62",
                          "quantity": "4",
                          "sku": "12116768",
                        },
                        Object {
                          "price": "223.71",
                          "quantity": "1",
                          "sku": "12216760",
                        },
                        Object {
                          "price": "73.16",
                          "quantity": "1",
                          "sku": "12516766",
                        },
                      ],
                      "SE": Array [
                        Object {
                          "price": "208.43",
                          "quantity": 1,
                          "sku": "12216760",
                        },
                        Object {
                          "price": "172.94",
                          "quantity": 4,
                          "sku": "12116768",
                        },
                        Object {
                          "price": "231.63",
                          "quantity": 1,
                          "sku": "12316762",
                        },
                        Object {
                          "price": "68.17",
                          "quantity": 1,
                          "sku": "12516766",
                        },
                      ],
                      "TX": Array [
                        Object {
                          "price": "231.63",
                          "quantity": "1",
                          "sku": "12316762",
                        },
                        Object {
                          "price": "208.43",
                          "quantity": "1",
                          "sku": "12216760",
                        },
                        Object {
                          "price": "68.17",
                          "quantity": "1",
                          "sku": "12516766",
                        },
                        Object {
                          "price": "172.94",
                          "quantity": "4",
                          "sku": "12116768",
                        },
                      ],
                    },
                    "price": 1199.99,
                    "primary_image": "http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc",
                    "shipping_cost_code": Object {
                      "FL": "ROOM",
                      "SE": "ROOM",
                      "TX": "ROOM",
                    },
                    "sku": "1141677P",
                    "title": "Laney Park Dark Gray 7 Pc Sectional",
                  },
                  "quantity": 3,
                  "sku": "1141677P",
                },
              ],
            }
          }
          discount={0}
          onlyButtons={true}
          productsAvailable={false}
          removeUnavailableItems={[Function]}
          showPayPal={false}
        />
      </div>
      <p
        className="financing-disclaimer cell small-12"
      >
        *when paying with a rooms to go credit card
      </p>
      <CreditCardBanner />
      <CondensedContact />
    </div>
    <div
      className="checkout-container cell small-12 large-2 grid-margin-y"
    >
      <CheckoutSticky
        cart={
          Object {
            "cartItems": Array [
              Object {
                "activeAddons": null,
                "price": 279.99,
                "product": Object {
                  "category": "tables",
                  "delivery_type": "O",
                  "free_shipping": true,
                  "list_price": 349.99,
                  "on_sale": true,
                  "price": 279.99,
                  "primary_image": "http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090",
                  "shipping_cost_code": Object {
                    "FL": "VD71",
                    "SE": "VD71",
                    "TX": "VD71",
                  },
                  "sku": "23122360",
                  "strikethrough": Object {
                    "FL": true,
                    "OM": true,
                    "SE": true,
                    "TX": true,
                  },
                  "title": "Adaire Brown Side Table",
                },
                "quantity": 1,
                "sku": "23122360",
              },
              Object {
                "activeAddons": null,
                "price": 1199.99,
                "product": Object {
                  "category": "livingroom",
                  "delivery_type": "D",
                  "free_shipping": false,
                  "list_price": 1299.99,
                  "on_sale": true,
                  "package_skus": Object {
                    "FL": Array [
                      Object {
                        "price": "248.65",
                        "quantity": "1",
                        "sku": "12316762",
                      },
                      Object {
                        "price": "185.62",
                        "quantity": "4",
                        "sku": "12116768",
                      },
                      Object {
                        "price": "223.71",
                        "quantity": "1",
                        "sku": "12216760",
                      },
                      Object {
                        "price": "73.16",
                        "quantity": "1",
                        "sku": "12516766",
                      },
                    ],
                    "SE": Array [
                      Object {
                        "price": "208.43",
                        "quantity": 1,
                        "sku": "12216760",
                      },
                      Object {
                        "price": "172.94",
                        "quantity": 4,
                        "sku": "12116768",
                      },
                      Object {
                        "price": "231.63",
                        "quantity": 1,
                        "sku": "12316762",
                      },
                      Object {
                        "price": "68.17",
                        "quantity": 1,
                        "sku": "12516766",
                      },
                    ],
                    "TX": Array [
                      Object {
                        "price": "231.63",
                        "quantity": "1",
                        "sku": "12316762",
                      },
                      Object {
                        "price": "208.43",
                        "quantity": "1",
                        "sku": "12216760",
                      },
                      Object {
                        "price": "68.17",
                        "quantity": "1",
                        "sku": "12516766",
                      },
                      Object {
                        "price": "172.94",
                        "quantity": "4",
                        "sku": "12116768",
                      },
                    ],
                  },
                  "price": 1199.99,
                  "primary_image": "http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc",
                  "shipping_cost_code": Object {
                    "FL": "ROOM",
                    "SE": "ROOM",
                    "TX": "ROOM",
                  },
                  "sku": "1141677P",
                  "title": "Laney Park Dark Gray 7 Pc Sectional",
                },
                "quantity": 3,
                "sku": "1141677P",
              },
            ],
          }
        }
        discount={0}
        isMobile={false}
        productsAvailable={false}
        removeUnavailableItems={[Function]}
        rightSticky={true}
        showPayPal={false}
      />
    </div>
  </div>
</div>
`;
